pkg/venafi/tpp/sshCertUtils.go

```diff
@@ -241,3 +242,125 @@ func convertToGenericRetrieveResponse(data *certificate.TppSshCertRetrieveRespon
 	return response
 
 }
+
+func getSshConfigUrl(key, value string) string {
+	var url string
+	query := fmt.Sprintf("%s=%s", key, value)
+	query = netUrl.PathEscape(query)
+	url = fmt.Sprintf("%s?%s", urlResourceSshCAPubKey, query)
+	return url
+}
+
+func RetrieveSshConfig(c *Connector, ca *certificate.SshCaTemplateRequest) (*certificate.SshConfig, error) {
+
+	var url string
+	if ca.DN != "" {
+		fullPath := getCaDNFullPath(ca.DN)
+		url = getSshConfigUrl("DN", fullPath)
+		fmt.Println("Retrieving the configured CA public key for template:", fullPath)
+	} else if ca.Guid != "" {
+		url = getSshConfigUrl("guid", ca.Guid)
+		fmt.Println("Retrieving the configured CA public key for GUID:", ca.Guid)
+	} else {
+		return nil, fmt.Errorf("CA template or GUID are not specified")
+	}
+
+	statusCode, status, body, err := c.request("GET", urlResource(url), nil)
+
+	if err != nil {
+		return nil, err
+	}
+	conf := certificate.SshConfig{}
+	switch statusCode {
+
+	case http.StatusOK, http.StatusAccepted:
+		conf.CaPublicKey = string(body)
+
+	default:
+		return nil, fmt.Errorf("error while retriving CA public key, error body:%s, status:%s and status code:%v", string(body), status, statusCode)
+	}
+
+	if c.accessToken != "" {
+		principals, err := RetrieveSshCaPrincipals(c, ca)
+		if err != nil {
+			return nil, err
+		}
+
+		conf.Principals = principals
+	} else {
+		fmt.Println("Skipping retrieval of Principals. No authentication data is provided.")
```

Comment:
In the SSH Protect UI, we use Default Principals. Maybe, it makes sense to be consistent in the terminology.
```suggestion
		fmt.Println("Skipping retrieval of Default Principals. No authentication data is provided.")
```

Reply:
fixed on the most recent commits